src/ansi-c/c_typecheck_base.cpp

```diff
@@ -868,6 +890,14 @@ void c_typecheck_baset::typecheck_declaration(
           assigns.swap(lambda);
         }
 
+        typecheck_spec_frees(code_type.frees());
+        for(auto &frees : code_type.frees())
+        {
+          lambda_exprt lambda{temporary_parameter_symbols, frees};
+          lambda.add_source_location() = frees.source_location();
+          frees.swap(lambda);
+        }
```

Comment:
Nitpick: we might want to throw a simple warning saying there is current no implementation for it.